File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--add-listobject-controls-to-worksheets_2.cs
        private void AddListObject()
        {
            Worksheet worksheet = Globals.Factory.GetVstoObject(
                Globals.ThisAddIn.Application.ActiveWorkbook.Worksheets[1]);

            Microsoft.Office.Tools.Excel.ListObject list1;
            Excel.Range cell = worksheet.Range["$A$1:$D$4"];
            list1 = worksheet.Controls.AddListObject(cell, "list1");
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--add-listobject-controls-to-worksheets_2.vb
    Private Sub AddListObject()
        Dim NativeWorksheet As Microsoft.Office.Interop.Excel.Worksheet =
            Globals.ThisAddIn.Application.ActiveWorkbook.Worksheets(1)

        Dim worksheet As Microsoft.Office.Tools.Excel.Worksheet =
            Globals.Factory.GetVstoObject(NativeWorksheet)

        Dim list1 As Microsoft.Office.Tools.Excel.ListObject
        Dim cell As Excel.Range = worksheet.Range("$A$1:$D$4")
        list1 = worksheet.Controls.AddListObject(cell, "MyListObject")
    End Sub